### DISCUSSION TITLE
Trying to update an entry field via php EVENT_BEFORE_SAVE_ENTRY

### DISCUSSION THREAD

**Discussion by kaduppg** [craftcms/cms#9792] [Q&A]
Hey guys, 
I've been going through something which might be silly, but dont know how to do it. I would like to save an entry, and then inside Module capture this element and then change its value like so:

```php
Event::on(
    Elements::class,
    Elements::EVENT_BEFORE_SAVE_ELEMENT,
    function (ElementEvent $event) {
        $event->element->setFieldValue('readingTime', 30);
    }
);
```

How can I change the value and save it for real? With this code i can change it but it doesnt update with the new value.

**Comment by brandonkelly** [comment]
That should work. Did you verify that `$event->element` is actually the element you were intending on editing? Entries, assets, Matrix blocks, users, categories, and tags are all elements (plus other plugin-supplied types). And entries come in multiple forms – unpublished drafts, provisional drafts, normal drafts, revisions, and normal entries.

**Reply by kaduppg** [reply]
Yeah that works! \o/ ...  the problem now is im trying get the text and calculate the reading time and then set the field but for some reason this is not working. 

```
Event::on(
            Elements::class,
            Elements::EVENT_BEFORE_SAVE_ELEMENT,
            function (ElementEvent $event) {
               
                $words = count(str_word_count($event->element['text'], 1));
                $readingTime = (int)ceil($words / 200); 

                $event->element->setFieldValue('readingTime',$readingTime);
            }
        );
```

**Reply by kaduppg** [reply]
PS: Inside the entry i've got a content block(Matrix) where the text is, so I would like to read the text inside the block and then update the field reading time which is outside the contentBlock.

**Reply by brandonkelly** [reply]
You would need to loop through the Matrix field where the text actually is. Something like

```php
use craft\elements\Entry;
use craft\events\ElementEvent;
use craft\helpers\ElementHelper;
use craft\services\Elements;
use yii\base\Event;

Event::on(
    Elements::class,
    Elements::EVENT_BEFORE_SAVE_ELEMENT,
    function (ElementEvent $event) {
        // Ignore if it's not an entry, or it's a draft or revisions
        if (!$event->element instanceof Entry::class || ElementHelper::isDraftOrRevision($event->element)) {
            return;
        }

        $words = 0;
        foreach ($event->element->myMatrixField as $block) {
            $words += str_word_count($block->myTextField);
        }
        $readingTime = (int)ceil($words / 200);
        $event->element->setFieldValue('readingTime', $readingTime);
    }
);
```
